tools/clang/test/SemaHLSL/attributes/wavesize_range.hlsl

```diff
@@ -22,7 +22,6 @@ void node01(DispatchNodeInputRecord<INPUT_RECORD> input) { }
 [NodeLaunch("broadcasting")]
 [NumThreads(1,1,1)]
 [NodeMaxDispatchGrid(32,1,1)]
-/* expected-error@+2{{Preferred WaveSize value 32 must be between 16 and 16}} */
```

Comment:
I guess we can only produce one error at a time now? Could we get a new case to verify the previous error?

Reply:
How useful are multiple errors on the same attribute?

The error coming from validation should highlight the first error that needs fixing.  Other errors might be moot or change once you fix the first issue.

For instance, in this case, you need to omit the max parameter to the attribute, which means preferred must also be omitted.  If we improve the diagnostics for cases like that, emitting additional errors is basically adding noise that distracts from the primary thing that must be addressed.

I could change the validation method to flag multiple issues, but that seems to be of limited value.

Note: We should still add more test cases here, partly because I identified some cases that wouldn't have been caught or handled very well in Sema by the previous version, and partly to make sure we independently test each invalid scenario

Reply:
If there are multiple legitimately wrong things (as in these examples) printing all the errors allows the user to have the full list of errors after a single compile. That’s definitely useful rather than having to compile multiple times to get and resolve the errors.